### DISCUSSION TITLE
SSR just the document head, and leave the actual content as an SPA

### DISCUSSION THREAD

**Discussion by Bobakanoosh** [vikejs/vike#1195] [Discussion] [answered]
I need to be able to set the meta tags of a page based on route, e.g. `/profile/john-doe`, but don't want to SSR the actual content of the page. I want it to go into SPA mode. So essentially Vike would just change the meta tags, and then send off the `.html` file just like a traditional vue-spa does with an empty `<div id="app"></div>`

Is this possible with Vike?

**Comment by brillout** [comment, marked as answer]
> Is this possible with Vike?

Yes.

You have full control over how the page is rendered to HTML and on the client-side.

**Comment by lourot** [comment]
FYI this is what is done already in vike-react, vike-vue and vike-solid.

In [this example](https://github.com/vikejs/vike-vue/blob/main/examples/basic/pages/%2Bconfig.h.ts#L10) we set a `Head` component. If we also set `ssr: false`, then only `Head` will be SSRendered but not the `Page`. This is achieved [here](https://github.com/vikejs/vike-vue/blob/main/vike-vue/renderer/onRenderHtml.ts)

You can do the same in your pure Vike project or directly scaffold a new project using `vike-{react,vue,solid}` by using https://batijs.github.io/

**Reply by Bobakanoosh** [reply]
Awesome, didn't know that `vike-vue` existed honestly must have missed that in the docs. Glad this works, will use this instead of Nuxt! Appreciate your response
